docs/app/Components/ComponentDoc/ComponentProps.js

```diff
@@ -23,7 +23,12 @@ export default class ComponentProps extends Component {
     }
   }
   defaultValueRenderer = (item) => {
-    const defaultValue = _.get(item, 'defaultValue.value')
+    let defaultValue = _.get(item, 'defaultValue.value')
+
+    if (_.startsWith(defaultValue, 'function ')) {
+      defaultValue = defaultValue.match(/^function(.*)\{/)[1].trim()
```

Comment:
You are half correct, this is to get the function name.  However, the defaultValue here is a json representation of the function from react-docgen.  The defaultValue is the actual function stringified.

See `"value": "function formSerializer(formNode) {\n  debug('formSerializer...`:

```json
"serializer": {
  "type": {
    "name": "func"
  },
  "required": false,
  "defaultValue": {
    "value": "function formSerializer(formNode) {\n  debug('formSerializer()')\n  const json = {}\n  // handle empty formNode ref\n  if (!formNode) return json\n\n  // ----------------------------------------\n  // Checkboxes\n  // Single: { name: value|bool        }\n  // Group:  { name: [value|bool, ...] }\n\n  _.each(formNode.querySelectorAll('input[type=\"checkbox\"]'), (node, index, arr) => {\n    const name = getNodeName(node)\n    const checkboxesByName = _.filter(arr, { name })\n\n    // single: (value|checked)\n    if (checkboxesByName.length === 1) {\n      json[name] = node.checked && node.value !== 'on' ? node.value : node.checked\n      debugSerializedResult(json, name, node)\n      return\n    }\n\n    // groups (checked): [value, ...]\n    if (!Array.isArray(json[name])) json[name] = []\n    if (node.checked) json[name].push(node.value)\n    debugSerializedResult(json, name, node)\n\n    // in dev, warn about multiple checkboxes with a default browser value of \"on\"\n    if (process.NODE_ENV !== 'production' && node.value === 'on') {\n      const errorMessage = [\n        \"Encountered a checkbox in a group with the default browser value 'on'.\",\n        'Each checkbox in a group should have a unique value.',\n        \"Otherwise, the checkbox value will serialize as ['on', ...].\",\n      ].join(' ')\n      console.error(errorMessage, node, formNode) // eslint-disable-line no-console\n    }\n  })\n\n  // ----------------------------------------\n  // Radios\n  // checked: { name: checked value }\n  // none:    { name: null }\n\n  _.each(formNode.querySelectorAll('input[type=\"radio\"]'), (node, index, arr) => {\n    const name = getNodeName(node)\n    const checkedRadio = _.find(arr, { name, checked: true })\n\n    if (checkedRadio) {\n      json[name] = checkedRadio.value\n    } else {\n      json[name] = null\n    }\n\n    debugSerializedResult(json, name, node)\n\n    // in dev, warn about radios with a default browser value of \"on\"\n    if (process.NODE_ENV !== 'production' && node.value === 'on') {\n      const errorMessage = [\n        \"Encountered a radio with the default browser value 'on'.\",\n        'Each radio should have a unique value.',\n        \"Otherwise, the radio value will serialize as { [name]: 'on' }.\",\n      ].join(' ')\n      console.error(errorMessage, node, formNode) // eslint-disable-line no-console\n    }\n  })\n\n  // ----------------------------------------\n  // Other inputs\n  // { name: value }\n\n  _.each(formNode.querySelectorAll('input:not([type=\"radio\"]):not([type=\"checkbox\"])'), (node) => {\n    const name = getNodeName(node)\n    json[name] = node.value\n    debugSerializedResult(json, name, node)\n  })\n\n  // ----------------------------------------\n  // Other inputs and text areas\n  // { name: value }\n\n  _.each(formNode.querySelectorAll('textarea'), (node) => {\n    const name = getNodeName(node)\n    json[name] = node.value\n    debugSerializedResult(json, name, node)\n  })\n\n  // ----------------------------------------\n  // Selects\n  // single:   { name: value }\n  // multiple: { name: [value, ...] }\n\n  _.each(formNode.querySelectorAll('select'), (node) => {\n    const name = getNodeName(node)\n\n    if (node.multiple) {\n      json[name] = _.map(_.filter(node.querySelectorAll('option'), 'selected'), 'value')\n    } else {\n      json[name] = node.value\n    }\n\n    debugSerializedResult(json, name, node)\n  })\n\n  return json\n}",
    "computed": false
  },
  "docBlock": {
    "description": "Called onSubmit with the form node that returns the serialized form object",
    "tags": []
  }
}
```